### PR TITLE
[Serve][Release][Part1] Enable tests to GCE

### PR DESCRIPTION
**Pull request by sihanwang41**
## Why are these changes needed?

Add release tests to GCE.

Long running failure succeed:
https://buildkite.com/ray-project/release-tests-pr/builds/34156#01875cb1-619c-4752-9710-26c5b95de5e6
serve_single_deployment_1k_noop_replica succeed:
https://buildkite.com/ray-project/release-tests-pr/builds/34271#01876741-acaa-4d2a-8b50-64cadf8b67d4
serve_multi_deployment_1k_noop_replica succeed:
https://buildkite.com/ray-project/release-tests-pr/builds/34273#018767c2-f14f-4da6-a5cd-ad04675caa99
serve_autoscaling_single_deployment succeed
https://buildkite.com/ray-project/release-tests-pr/builds/34296#018768f2-9202-4590-8536-8454b60eaca5
serve_autoscaling_multi_deployment succeed
https://buildkite.com/ray-project/release-tests-pr/builds/34298#0187692f-f93c-4708-9ac9-acd32c9633ab

## Checks

- [ ] I've signed off every commit(by using the -s flag, i.e., `git commit -s`) in this PR.
- [ ] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://docs.ray.io/en/master/.
    - [ ] I've added any new APIs to the API Reference. For example, if I added a 
           method in Tune, I've added it in `doc/source/tune/api/` under the 
           corresponding `.rst` file.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [ ] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

### CODE REVIEW

**Review by edoakes — APPROVED** [review]

File: release/serve_tests/compute_tpl_32_cpu_autoscaling_gce.yaml [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+head_node_type:
+    name: head_node
+    instance_type: n2-standard-32 # m5.8xlarge
+
+worker_node_types:
+    - name: worker_node
+      instance_type: n2-standard-32 # m5.8xlarge
+      min_workers: 5
+      max_workers: 35
+      use_spot: false
```
Comment (can-anyscale) [inline release/serve_tests/compute_tpl_32_cpu_autoscaling_gce.yaml:17]: new lin?

File: release/serve_tests/compute_tpl_32_cpu_gce.yaml [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+head_node_type:
+    name: head_node
+    instance_type: n2-standard-32 # m5.8xlarge
+
+worker_node_types:
+    - name: worker_node
+      instance_type: n2-standard-32 # m5.4xlarge
+      min_workers: 32
+      max_workers: 32
+      use_spot: false
```
Comment (can-anyscale) [inline release/serve_tests/compute_tpl_32_cpu_gce.yaml:17]: nit: new line

File: release/serve_tests/compute_tpl_8_cpu_autoscaling_gce.yaml [hunk 3]
```diff
@@ -0,0 +1,17 @@
...
+head_node_type:
+    name: head_node
+    instance_type: c2-standard-8 # m5.2xlarge
+
+worker_node_types:
+    - name: worker_node
+      instance_type:  c2-standard-8 # m5.2xlarge
+      min_workers: 5
+      max_workers: 130
+      use_spot: false
```
Comment (can-anyscale) [inline release/serve_tests/compute_tpl_8_cpu_autoscaling_gce.yaml:17]: nit: new line

**Review by can-anyscale — APPROVED** [review]
gogogo

**Pull request merged**